### PR TITLE
fix: do not return updated messages from markAllRead if not needed

### PR DESCRIPTION
**Pull request by Bilb**

### CODE REVIEW

File: ts/data/data.ts [hunk 1]
```diff
@@ -493,9 +493,17 @@ async function getUnreadByConversation(conversationId: string): Promise<MessageC
 }
 
 async function markAllAsReadByConversationNoExpiration(
-  conversationId: string
+  conversationId: string,
+  returnMessagesUpdated: boolean // for performances reason we do not return them because usually they are not needed
```
Comment (yougotwill) [inline ts/data/data.ts:497]: * performance reasons

File: ts/node/database_utility.ts [hunk 2]
```diff
@@ -264,7 +264,7 @@ export function rebuildFtsTable(db: BetterSqlite3.Database) {
           CREATE TRIGGER messages_on_delete AFTER DELETE ON ${MESSAGES_TABLE} BEGIN
             DELETE FROM ${MESSAGES_FTS_TABLE} WHERE id = old.id;
           END;
-          CREATE TRIGGER messages_on_update AFTER UPDATE ON ${MESSAGES_TABLE} BEGIN
+          CREATE TRIGGER messages_on_update AFTER UPDATE ON ${MESSAGES_TABLE} WHEN new.body <> old.body BEGIN
```
Comment (yougotwill) [inline ts/node/database_utility.ts:267]: Just to confirm. Is this making sure that the search triggers have been updated only when the body of a message has changed?

**Review by yougotwill — CHANGES_REQUESTED** [review]

File: ts/node/database_utility.ts (hunk 2 above)
Comment (Bilb) [reply ts/node/database_utility.ts:267]: Not sure what you meant exactly, but yeah, it is meant to only rerun the triggers for the full text search logic when the body has changed and not anything else (i.e., marking a message as read does not update the body and so does not need to trigger a refresh of the full text search table)

File: ts/data/data.ts (hunk 1 above)
Comment (Bilb) [reply ts/data/data.ts:497]: fioxed

File: ts/node/database_utility.ts (hunk 2 above)
Comment (yougotwill) [reply ts/node/database_utility.ts:267]: Yeah, I'm probably using the wrong terminology but you answered my question. Thank you 🙏

**Review by yougotwill — APPROVED** [review]
LGTM 🚀

**Pull request merged**